File: 프로그래머스/2/42586. 기능개발/기능개발.js
// 스택,큐 & 그리디
function solution(progresses, speeds) {
    var answer = [];
    const days = progresses.map((p,index)=> Math.ceil((100-p)/speeds[index]));
    
    let maxDay = days[0];
    let count = 1;
    
    for(let i=1; i<days.length; i++){
        if(maxDay<days[i]){
            maxDay = days[i];
            answer.push(count);
            count = 1;
            continue;
        }
        count++;
    }
    answer.push(count);
    
    return answer;
}

File: 프로그래머스/2/42586. 기능개발/기능개발.py
import math

def solution(progresses, speeds):
    answer = []
    
    max_progress = math.ceil((100 - progresses[0]) / speeds[0])
    count = 0
    
    for i in range(len(progresses)):
        progress = math.ceil((100 - progresses[i]) / speeds[i])
        
        if max_progress >= progress:
            count += 1
        else:
            answer.append(count)
            count = 1
            max_progress = progress
            
    if count != 0:
        answer.append(count)
    
    return answer